rust/worker/src/cache/cache.rs

```diff
@@ -0,0 +1,190 @@
+use crate::cache::config::CacheConfig;
+use crate::config::Configurable;
+use crate::errors::ChromaError;
+use async_trait::async_trait;
+use core::hash::Hash;
+use foyer::Cache as FoyerCache;
+use foyer::CacheBuilder;
+use foyer::LfuConfig;
+use foyer::LruConfig;
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use std::sync::Arc;
+use thiserror::Error;
+
+#[derive(Clone)]
+pub(crate) enum Cache<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
```

Comment:
nit - can we abstract out Send+Sync+Hash+Eq into a trait bound for reuse?

Reply:
I will address this in a follow-up PR.